src/server.c

```diff
@@ -5081,7 +5081,6 @@ int redisFork(int purpose) {
         if (childpid == -1) {
             return -1;
         }
-        updateDictResizePolicy();
```

Comment:
@oranagra Initially, I called `dictEnableResize` in `redisFork` in my PR, but I thought  we create `updateDictResizePolicy` to avoid calling `dictEnableResize` and `dictDisableResize` directly. If we pass an argument to `updateDictResizePolicy`, why not call them directly.
Currently, `updateDictResizePolicy` in `checkChildrenDone` is fine.

Reply:
you're right.
however, in the past the logic of `updateDictResizePolicy` was more complicated, now it's just uses `hasActiveChildProcess`, and anyway, the conditions inside it, are coupled with the triggers to update it. i.e. if we add more conditions, we need to call it in more places.

i'm just looking for a way to reduce the extra steps needed after each call to `redisFork` so that something won't be forgotten.

one additional idea is to add a server struct member like `child_pid` (without specifying it's purpose), if we do that we can set it inside `redisFork`, and check for it in `updateDictResizePolicy`, so that we don't need to move that call to the caller of `redisFork`.
but what i don't like here is the redundant info (two members storing the same pid value).

so maybe a way to solve it is to unify `rdb_child_pid`, `aof_child_pid`, and `module_child_pid` into one `child_pid`, and add a member named `current_fork_child`, with similar value to `in_fork_child` (i.e. `CHILD_TYPE_xxx`).

then redisFork handles everything, and the caller doesn't even need to store the pid.
we'll also have to add another helper method like `childExited` and call it from `killAppendOnlyChild` and alike.

what do you think of this cleanup?

if you don't like it, we can keep your changes, and possibly just add something in serverCron with a comment saying it's just safety.

Reply:
In `redisFork`, we needn't disable dict rehash when fork for lua just like before.

> so maybe a way to solve it is to unify `rdb_child_pid`, `aof_child_pid`, and `module_child_pid` into one `child_pid`, and add a member named `current_fork_child`, with similar value to `in_fork_child` (i.e. `CHILD_TYPE_xxx`).

Hi @oranagra I understand you want to put all related to fork into `redisFork` and I agree. For cleanup, I consider it's a good idea, but we need to do much work carefully now, I think we can refactor it at the right opportunity(need to add new things related to `fork`, or new big version) instead of now

Reply:
the Lua (LDB), was probably excluded from many of these fork aspects just because it was:
1. not very important since not meant to use in production.
2. it was hard to follow all the fork requirements since they were spread all over the place, but now that's being refactored into a common code, it's easy.

I think we **are** working now on a big version (6.2), unstable already contains massive changes that are not gonna make it to 6.0.
One possibility is to push your simple fix as one commit (to be cherry picked), and then in an immediate follow up do the refactory.
But considering the area we're gonna touch didn't probably change between unstable and 6.0, i think we can do the refactory and cherry pick that to 6.0?

This refactory seems rather stright forward. not sure it adds any risk. we just avoid saving the pid into different variables per fork purpose, and instead save just one pid, and a variable that says what's the fork purpose.

if you disagree, then let's take the safe way, merge this bugfix, and do the cleanup / refactory after.
let me know.

Reply:
Hi@oranagra Yes, we are working for 6.2. I agree we refactor in 6.2 and make somethings clear and less error-prone.
But now in my opinion, I hope we can take the safe way, merge this bugfix, and refactor later. Moreover, we also need to take it to 6.0, so I think this way is more safe.

I submit a commit that updateDictResizePolicy in serverCron.